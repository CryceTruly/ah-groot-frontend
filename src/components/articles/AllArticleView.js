import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

export class AllArticleView extends Component {
  constructor(props) {
    super(props);
    let { history } = this.props;
  }
  render() {
    const { title, description, slug } = this.props;
    return (
      <div className="col s12 custom">
        <div className="card grey lighten-5">
          <div
            slug={slug}
            className="card-content black-text"
            onClick={event => {
              const slug = event.currentTarget.getAttribute('slug');
              this.props.history.push(`/article/${slug}/view`);
            }}
          >
            <span className="card-title truncate">{title}</span>
            <span className="card-description truncate">{description}</span>
          </div>
          <div className="card-action">
            <button >1 like</button>
            <button >3 dislikes</button>
            <button >0 comments</button>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(AllArticleView);
