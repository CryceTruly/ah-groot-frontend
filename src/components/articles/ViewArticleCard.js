import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import endPoints, {authHeader} from '../../containers/urls';
import axios from 'axios';

export class ArticleView extends Component {
  constructor(props) {
    super(props);
    let { history } = this.props;
  }

  deleteArticle = slug => {
    axios
      .delete(`${endPoints.singleArticle}${slug}/`, authHeader)
      .then(function(response) {
        if (response.status === 204) {
          M.toast({ html: 'This article has been deleted ', classes: 'green' });
          Window.location.reload();
        }
      })
      .catch(function(error) {
        return;
      });
  };

  editArticle = slug => {
    this.props.history.push(`/me/article/${slug}/edit`);
  }

  render() {
    const { title, description, slug } = this.props;
    return (
      <div className="col s12 custom">
        <div className="card grey lighten-5">
          <div className="card-content black-text">
            <span className="card-title truncate">{title}</span>
            <span className="card-description truncate">{description}</span>
          </div>
          <div className="card-action">
            <span slug={slug} className='edit-artilce' onClick={() => this.editArticle(slug)}>
              <i className="material-icons">edit</i>
            </span>
            <span slug={slug} className='delete-artilce'onClick={() => this.deleteArticle(slug)}>
            <i className="material-icons">delete</i>
            </span>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(ArticleView);
