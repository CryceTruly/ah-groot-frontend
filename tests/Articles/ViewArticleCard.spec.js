import React from 'react';
import { shallow } from 'enzyme';
import moxios from 'moxios';

import { AllArticleView } from '../../src/components/articles/AllArticleView';
import { ArticleView } from '../../src/components/articles/ViewArticleCard';

describe('AllArticleView', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<AllArticleView />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ArticleView', () => {
  beforeEach(function() {
    moxios.install();
  });

  afterEach(function() {
    moxios.uninstall();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(<ArticleView />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without call the edit function', () => {
    const wrapper = shallow(<ArticleView />);
    wrapper.setProps({history: { push: jest.fn()}});
    wrapper.instance().editArticle("slug");
  });

  it('should render without call the delete function', () => {
    const wrapper = shallow(<ArticleView />);
    wrapper.instance().deleteArticle("slug");
    moxios.wait(() => {
      const requestM = moxios.requests.mostRecent();
      requestM.respondWith({
        status: 204,
        response: {},
      });
    });
  });
});

