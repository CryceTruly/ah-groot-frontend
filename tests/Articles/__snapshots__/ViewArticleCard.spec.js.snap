// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllArticleView should render without crashing 1`] = `ShallowWrapper {}`;

exports[`ArticleView should render without crashing 1`] = `ShallowWrapper {}`;
